test/e2e/wireguard_test.go

```diff
@@ -127,3 +129,62 @@ func testServiceConnectivity(t *testing.T, data *TestData) {
 	err = data.runNetcatCommandFromTestPod(clientPodName, testNamespace, clientPodNodeIP, svc.Spec.Ports[0].NodePort)
 	require.NoError(t, err, "Pod %s should be able to connect the service's NodePort %s:%d, but was not able to connect", clientPodName, clientPodNodeIP, svc.Spec.Ports[0].NodePort)
 }
+
+// testRoute make sure that the route to Pod on peer Node or route to peer gateway is targeting the WireGuard device.
+func testRoute(t *testing.T, data *TestData) {
+	srcNode := nodeName(0)
+	var srcNodeIP, peerGatewayIP, peerPodIP string
+	IPv4, IPv6 := nodeGatewayIPs(0)
+	if IPv4 != "" {
+		srcNodeIP = IPv4
+	} else {
+		srcNodeIP = IPv6
+	}
+	IPv4, IPv6 = nodeGatewayIPs(1)
+	if IPv4 != "" {
+		peerGatewayIP = IPv4
+	} else {
+		peerGatewayIP = IPv6
+	}
+
+	peerPod := randName("agnhost-")
+	require.NoError(t, data.createAgnhostPodOnNode(peerPod, testNamespace, nodeName(1), false))
+	defer data.deletePodAndWait(defaultTimeout, peerPod, testNamespace)
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, peerPod, testNamespace))
+
+	peerPodIPs, err := data.podWaitForIPs(defaultTimeout, peerPod, testNamespace)
+	require.NoError(t, err)
+	if peerPodIPs.ipv4 != nil {
+		peerPodIP = peerPodIPs.ipv4.String()
+	} else {
+		peerPodIP = peerPodIPs.ipv6.String()
+	}
+
+	tests := []struct {
+		name               string
+		dstIP              string
+		expectedDeviceName string
+		expectedSrcIP      string
+	}{
+		{
+			name:               "toPodOnPeerNode",
+			dstIP:              peerPodIP,
+			expectedDeviceName: "antrea-wg0",
+			expectedSrcIP:      srcNodeIP,
+		},
+		{
+			name:               "toPeerGateway",
+			dstIP:              peerGatewayIP,
+			expectedDeviceName: "antrea-wg0",
+			expectedSrcIP:      srcNodeIP,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cmd := fmt.Sprintf("ip route get %v", tt.dstIP)
+			_, stdout, _, err := data.RunCommandOnNode(srcNode, cmd)
```

Comment:
as a rule I think it is better to run commands from Pods if possible, in this case from the antrea-agent Pod. There is no real guarantee that the `ip` utility is available on Nodes in an arbitrary cluster.

Reply:
Initially i tried to run command from Pod but got following error :-
`Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "e1ab22552b29f6751dd1b924a33c076a5201d5f83fb002cf1316a725cca7197f": OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "ip route get 10.10.1.105": executable file not found in $PATH: unknown`
So, switched to Node instead of Pod. I am using kind cluster on M1 Mac.

Reply:
Your Antrea image must have been wrong, as it sounds very similar to https://github.com/antrea-io/antrea/issues/3346. The base image was built for the wrong architecture.

Reply:
I built the Antrea image using `DOCKER_BUILDKIT=1 ./hack/build-antrea-linux-all.sh --platform linux/arm64`, still got same error. 
And, other tests are running fine, if it was due to wrong Antrea image then other tests like ProxyTest( in which some command required to be run from Antrea agent pod) must fail with same error, but those tests are running fine.

Reply:
Yes, your code is wrong. It should be:
```go
cmd := []string{"ip", "route" "get", tt.dstIP)}
```
and not:
```go
cmd := []string{fmt.Sprintf("ip route get %v", tt.dstIP)}
```
In the second version (what you have in your code), `ip route get 10.10.1.105` is interpreted as the name of the executable.

Reply:
Thank you @antoninbas